File: readings/templates/template.py
"""
算法竞赛 Python 通用模板
比赛时直接复制粘贴此文件内容，在此基础上写题解

注意：Python 在 CPL 系统中可能会遇到递归深度限制和时间限制问题。
      如果题目数据量较大 (n > 10^5 的复杂操作)，建议使用 C++。
      但对于输入规模较小、或字符串处理为主的题目，Python 效率更高。
"""

import sys
import math
import itertools
import collections
import heapq
import bisect
import functools
import random
from typing import List, Tuple, Set, Dict, Optional

# ============ 输入输出优化 ============
# 使用 sys.stdin.read() 一次性读入所有内容，自己分割
# 这是 Python 在算法竞赛中最快的输入方式

def input():
    return sys.stdin.readline().rstrip('\n')

def ints() -> List[int]:
    """读取一行整数，返回列表"""
    return list(map(int, input().split()))

# 如果需要更快（万级以上数据），使用 sys.stdin.buffer.read()
# data = sys.stdin.buffer.read().split()
# n = int(data[0]); arr = list(map(int, data[1:]))

# ============ 常用函数别名 ============
gcd = math.gcd       # Python 3.5+ 自带 math.gcd
lcm = math.lcm       # Python 3.9+ 自带 math.lcm
isqrt = math.isqrt   # 整数平方根

# ============ 数学工具函数 ============

def lcm_custom(a: int, b: int) -> int:
    """最小公倍数 (Python < 3.9 的回退)"""
    return a // math.gcd(a, b) * b

def qpow(a: int, b: int, m: int = 10**9 + 7) -> int:
    """快速幂 a^b % m"""
    res = 1
    a %= m
    while b:
        if b & 1:
            res = res * a % m
        a = a * a % m
        b >>= 1
    return res

def ceil_div(a: int, b: int) -> int:
    """向上取整除法"""
    return (a + b - 1) // b

def is_prime(n: int) -> bool:
    """判断质数 O(sqrt(n))"""
    if n < 2:
        return False
    i = 2
    while i * i <= n:
        if n % i == 0:
            return False
        i += 1
    return True

def sieve(n: int) -> List[int]:
    """埃氏筛法：返回 [2, n] 内所有质数"""
    is_p = [True] * (n + 1)
    is_p[0] = is_p[1] = False
    for i in range(2, int(n ** 0.5) + 1):
        if is_p[i]:
            for j in range(i * i, n + 1, i):
                is_p[j] = False
    return [i for i in range(2, n + 1) if is_p[i]]

# ============ 常用数据结构速查 ============
# 以下列出 Python 内置数据结构的核心用法，详见 day1 文件

# --- list (动态数组，对应 C++ vector) ---
# arr = []
# arr.append(x)          # 末尾添加
# arr.pop()              # 末尾删除
# arr.sort()             # 升序排序
# arr.sort(reverse=True) # 降序排序
# arr.sort(key=lambda x: (x[0], -x[1]))  # 自定义排序

# --- dict (哈希表，对应 C++ unordered_map) ---
# d = {}
# d[key] = value
# d.get(key, default)    # 安全取值
# collections.Counter(iterable)   # 计数专用

# --- set (哈希集合，对应 C++ unordered_set) ---
# s = set()
# s.add(x); s.remove(x); x in s

# --- 有序字典/集合 (对应 C++ map/set) ---
# Python 标准库无内置平衡树，需要排序时转为 list 再 sort

# --- 堆 (对应 C++ priority_queue) ---
# import heapq
# pq = []
# heapq.heappush(pq, x)       # 小顶堆
# heapq.heappop(pq)
# # 大顶堆：存负数 heapq.heappush(pq, -x); y = -heapq.heappop(pq)

# --- 双端队列 (对应 C++ deque) ---
# from collections import deque
# dq = deque()
# dq.append(x); dq.popleft(); dq.appendleft(x)

# --- 二分查找 (对应 C++ lower_bound) ---
# import bisect
# pos = bisect.bisect_left(arr, x)   # 第一个 >= x 的位置
# pos = bisect.bisect_right(arr, x)  # 第一个 > x 的位置

# ============ 递归深度设置 ============
# 如果使用 DFS 递归，务必设置较大的递归深度
sys.setrecursionlimit(1000000)

# ============ 模数相关常量 ============
MOD = 10**9 + 7
INF = 10**18

# ============ 核心：每题的 solve 函数 ============
def solve():
    """在此编写每道题的解法"""
    pass

# ============ 主程序 ============
if __name__ == '__main__':
    # 单组数据
    # solve()

    # 多组数据：第一行是测试组数 T
    t = int(input())
    for _ in range(t):
        solve()

File: readings/templates/template.cpp
/**
 * 算法竞赛 C++ 通用模板
 * 比赛时直接复制粘贴此文件内容，在此基础上写题解
 *
 * 使用说明：
 * 1. 将 solve() 函数中的逻辑替换为每道题的解法
 * 2. 多组测试数据时保留 while(t--) 结构
 * 3. 单组数据时删除 while(t--) 即可
 */

#include <bits/stdc++.h>
using namespace std;

// ============ 类型别名 (加快手速) ============
using ll = long long;
using ull = unsigned long long;
using pii = pair<int, int>;
using pll = pair<ll, ll>;
using vi = vector<int>;
using vll = vector<ll>;
using vvi = vector<vi>;
using vvll = vector<vll>;

// ============ 常用常量 ============
const int INF = 0x3f3f3f3f;          // 约 1.06e9，两倍不会溢出
const ll LINF = 0x3f3f3f3f3f3f3f3fLL; // 约 4.5e18
const int MOD = 1e9 + 7;              // 常见模数，或者 998244353
const double EPS = 1e-9;              // 浮点数比较误差

// ============ 常用宏 ============
#define sz(x) ((int)(x).size())
#define all(x) (x).begin(), (x).end()
#define rall(x) (x).rbegin(), (x).rend()
#define fi first
#define se second
#define pb push_back
#define eb emplace_back
#define rep(i, a, b) for (int i = (a); i < (b); ++i)
#define per(i, a, b) for (int i = (b) - 1; i >= (a); --i)
#define each(i, x) for (auto &i : x)

// ============ 调试宏 (提交前删除或注释掉) ============
// #define DEBUG
#ifdef DEBUG
#define debug(x) cerr << #x << " = " << (x) << endl
#define debug2(x, y) cerr << #x << " = " << (x) << ", " << #y << " = " << (y) << endl
#else
#define debug(x)
#define debug2(x, y)
#endif

// ============ 数学工具函数 ============

// 最大公约数 (C++17 起可直接用 std::gcd)
ll gcd(ll a, ll b) { return b ? gcd(b, a % b) : a; }

// 最小公倍数 (注意先除后乘防溢出)
ll lcm(ll a, ll b) { return a / gcd(a, b) * b; }

// 快速幂 a^b % m  — 几乎所有取模幂运算都用它
ll qpow(ll a, ll b, ll m = MOD) {
    ll res = 1;
    a %= m;
    while (b) {
        if (b & 1) res = res * a % m;
        a = a * a % m;
        b >>= 1;
    }
    return res;
}

// 向上取整除法 (a+b-1)/b，或使用此函数
ll ceil_div(ll a, ll b) { return (a + b - 1) / b; }

// 判断是否为质数 (sqrt(n) 复杂度，用于 n <= 1e12)
bool is_prime(ll n) {
    if (n < 2) return false;
    for (ll i = 2; i * i <= n; ++i)
        if (n % i == 0) return false;
    return true;
}

// 埃氏筛法：返回 [2, n] 内所有质数
vi sieve(int n) {
    vector<bool> is_p(n + 1, true);
    vi primes;
    is_p[0] = is_p[1] = false;
    for (int i = 2; i <= n; ++i) {
        if (!is_p[i]) continue;
        primes.pb(i);
        for (ll j = (ll)i * i; j <= n; j += i)
            is_p[j] = false;
    }
    return primes;
}

// ============ 常用 STL 用法速查 ============
// 见 day1 详细说明，这里只放最常用的模式：

// --- 排序 ---
// sort(all(v));                          // 升序
// sort(all(v), greater<int>());          // 降序
// sort(all(v), [](auto &a, auto &b) {    // 自定义
//     return a.fi != b.fi ? a.fi < b.fi : a.se > b.se;
// });

// --- 去重 (需先排序) ---
// v.erase(unique(all(v)), v.end());

// --- 二分查找 (有序数组) ---
// int pos = lower_bound(all(v), x) - v.begin();   // 第一个 >= x 的位置
// int pos = upper_bound(all(v), x) - v.begin();   // 第一个 > x 的位置
// bool exist = binary_search(all(v), x);

// --- map/unordered_map ---
// unordered_map<int, int> cnt;  // 哈希表，O(1)，无序
// map<int, int> mp;             // 红黑树，O(log n)，有序

// --- set/unordered_set ---
// unordered_set<int> st;        // 哈希集合，O(1)
// set<int> st;                  // 有序集合，O(log n)

// --- 优先队列 ---
// priority_queue<int> pq;                          // 大顶堆 (默认)
// priority_queue<int, vi, greater<int>> min_pq;     // 小顶堆
// priority_queue<pii> pq;                          // 按 first 排序的大顶堆

// --- string 常用 ---
// string s = to_string(123);      // 数字转字符串
// int x = stoi("123");            // 字符串转数字
// ll x = stoll("1234567890123");  // 字符串转 long long
// auto pos = s.find("ab");        // 查找子串，返回位置或 string::npos
// s.substr(pos, len);             // 截取子串
// reverse(all(s));                // 反转

// ============ 输入输出优化 ============
void fast_io() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    // 注意：使用 fast_io 后不要混用 scanf/printf 和 cin/cout
    // 如果题目输入量特别大（>10^6 个数），可以加：
    // cin.ignore(numeric_limits<streamsize>::max(), '\n');
}

// ============ 核心：每题的 solve 函数 ============
void solve() {
    // 在此编写每道题的解法
    // 多组数据时：

    // int t;
    // cin >> t;
    // while (t--) {
    //     // 每组数据的逻辑
    // }
}

// ============ 主函数 ============
int main() {
    fast_io();
    // 如果只有一组数据，直接用 solve();
    // 多组数据在 solve 内部处理

    int t = 1;
    // cin >> t;  // 取消注释以支持多组数据
    while (t--) solve();

    return 0;
}
